### PR TITLE
Fix oneshot toggle logic

### PR DESCRIPTION
**Pull request by drashna**
## Description

#16617  tried to fix the bug, but honestly made it more convoluted, and broke it further.  

This changes the variable name and simplifies the logic, so it's clear and concise. 

## Types of Changes

- [x] Core
- [x] Bugfix

## Issues Fixed or Closed by This PR

* Oneshot toggle and oneshot keys broken.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/keycode_config.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ typedef union {
         bool nkro : 1;
         bool swap_lctl_lgui : 1;
         bool swap_rctl_rgui : 1;
-        bool oneshot_disable : 1;
+        bool oneshot_enable : 1;
```
Comment (fauxpark) [inline quantum/keycode_config.h:40]: I would guess EEPROM now needs to init to 1 here?

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: quantum/keycode_config.h (hunk 1 above)
Comment (drashna) [reply quantum/keycode_config.h:40]: yup. commit incoming for that

**Comment by drashna** [comment]
Honestly, I'd prefer a lighter touch, but fixing the function created some very convoluted logic.  And just made it harder to read. 

we could go with that, but I think this would be more maintainable then negatives and double negatives, etc.

**Comment by zvecr** [comment]
You would also need to bump the eeprom version given the value is now inverted.

**Review by zvecr — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**